providers/unleash/src/main/java/dev/openfeature/contrib/providers/unleash/UnleashProvider.java

```diff
@@ -0,0 +1,173 @@
+        super.emitProviderReady(details);
+        state = ProviderState.READY;
+    }
+
+    @Override
+    public void emitProviderError(ProviderEventDetails details) {
+        super.emitProviderError(details);
+        state = ProviderState.ERROR;
+    }
+
+    @Override
+    public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        UnleashContext context = ctx == null ? UnleashContext.builder().build() : ContextTransformer.transform(ctx);
+        boolean featureBooleanValue = unleash.isEnabled(key, context, defaultValue);
+        return ProviderEvaluation.<Boolean>builder()
+            .value(featureBooleanValue)
+            .build();
+    }
+
+    @Override
+    public ProviderEvaluation<String> getStringEvaluation(String key, String defaultValue, EvaluationContext ctx) {
+        ProviderEvaluation<Value> valueProviderEvaluation = getObjectEvaluation(key, new Value(defaultValue), ctx);
+        return ProviderEvaluation.<String>builder()
+            .value(valueProviderEvaluation.getValue().asString())
+            .variant(valueProviderEvaluation.getVariant())
+                .errorCode(valueProviderEvaluation.getErrorCode())
+                .reason(valueProviderEvaluation.getReason())
+                .flagMetadata(valueProviderEvaluation.getFlagMetadata())
+            .build();
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(String key, Integer defaultValue, EvaluationContext ctx) {
+        throw new TypeMismatchError(NOT_IMPLEMENTED);
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(String key, Double defaultValue, EvaluationContext ctx) {
+        throw new TypeMismatchError(NOT_IMPLEMENTED);
+    }
+
+    @Override
+    public ProviderEvaluation<Value> getObjectEvaluation(String key, Value defaultValue, EvaluationContext ctx) {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError(PROVIDER_NOT_YET_INITIALIZED);
+            }
+            throw new GeneralError(UNKNOWN_ERROR);
+        }
+        UnleashContext context = ctx == null ? UnleashContext.builder().build() : ContextTransformer.transform(ctx);
+        Variant evaluatedVariant = unleash.getVariant(key, context);
+        String variantName;
+        Value value;
+        if (DISABLED_VARIANT.equals(evaluatedVariant)) {
+            variantName = null;
+            value = defaultValue;
+        } else {
+            variantName = evaluatedVariant.getName();
+            value = evaluatedVariant.getPayload().map(p -> new Value(p.getValue())).orElse(null);
```

Comment:
I think we need to rethink the value part a bit. By just mapping out the payload value we loose a few important details
* payload type (make it possible to parse the value)
* Is the feature enabled

Reply:
I added some details to evaluation via [flag metadata](https://github.com/open-feature/java-sdk-contrib/compare/main...liran2000:java-sdk-contrib:feature/unleash?expand=1#diff-33b37a3e9b9bd14c4d0853c582e47b468d3927d41da53b47900a8c60a90b1133R47):

Additional evaluation data can be received via flag metadata, such as:

enabled - boolean
variant-stickiness - string
payload-type - string, optional